src/dmd/clone.d

```diff
@@ -536,16 +525,14 @@ FuncDeclaration buildXopEquals(StructDeclaration sd, Scope* sc)
         if (FuncDeclaration fd = eq.isFuncDeclaration())
         {
             TypeFunction tfeqptr;
-            {
-                Scope scx;
-                /* const bool opEquals(ref const S s);
-                 */
-                auto parameters = new Parameters();
-                parameters.push(new Parameter(STC.ref_ | STC.const_, sd.type, null, null, null));
-                tfeqptr = new TypeFunction(ParameterList(parameters), Type.tbool, LINK.d);
-                tfeqptr.mod = MODFlags.const_;
-                tfeqptr = cast(TypeFunction)tfeqptr.typeSemantic(Loc.initial, &scx);
-            }
+            Scope scx;
```

Comment:
this change has some merit, but line 528 should be moved down to just before 535.